types/react/index.d.ts

```diff
@@ -101,6 +101,9 @@ declare namespace React {
      */
     type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
 
+    /**
+     * @internal
+     */
     type JSXElementConstructor<P> =
```

Comment:
I figured that marking this as `@internal` would discourage people from using it. I can't quite see a use case that `ComponentType` doesn't cover - but I would love to know more about why this type exists and whether it can be useful in userland.

Reply:
I too would love to know

Reply:
It's the thiner type than `ComponentType` which can be very annoying to deal with in assignability and error messages because it takes all the statics with it. In most case `JSXElementConstructor<Props>` is the thing you actually want when you accept components.

Has a terrible name though. But definitely not internal. After 19 we should revisit this since a lot of the (from a type perspective) expensive statics will be removed anyway.

Reply:
Gooootcha, so it doesn't also receive `.displayName`, `.defaultProps` etc. Thanks, I can work with that.